cirq-core/cirq/transformers/target_gatesets/sqrt_iswap_gateset.py

```diff
@@ -0,0 +1,281 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Target gateset used for compiling circuits to √iSWAP + 1-q rotations + measurement gates."""
+
+from typing import Any, Dict, Optional, TYPE_CHECKING
+
+import numpy as np
+import sympy
+from cirq import ops, protocols
+from cirq.transformers.analytical_decompositions import two_qubit_to_sqrt_iswap
+from cirq.transformers.target_gatesets import compilation_target_gateset
+
+if TYPE_CHECKING:
+    import cirq
+
+
+class SqrtIswapTargetGateset(compilation_target_gateset.TwoQubitCompilationTargetGateset):
+    """Target gateset containing √iSWAP + single qubit rotations + Measurement gates."""
+
+    def __init__(
+        self,
+        *,
+        atol: float = 1e-8,
+        required_sqrt_iswap_count: Optional[int] = None,
+        use_sqrt_iswap_inv: bool = False,
+    ):
+        """Initializes `cirq.SqrtIswapTargetGateset`
+
+        Args:
+            atol: A limit on the amount of absolute error introduced by the decomposition.
+            required_sqrt_iswap_count: When specified, the `decompose_to_target_gateset` will
+                decompose each operation into exactly this many sqrt-iSWAP gates even if fewer is
+                possible (maximum 3). A ValueError will be raised if this number is 2 or lower and
+                synthesis of the operation requires more.
```

Comment:
Would an error also be raised if the number is 1 or lower and the operation synthesis requires more ?

Reply:
Yes.